examples/platforms/da15000/radio.c

```diff
@@ -0,0 +1,562 @@
+
+    while (REG_GETF(CRG_TOP, SYS_STAT_REG, FTDF_IS_UP) == 0x0);
+
+    REG_CLR_BIT(CRG_TOP, PMU_CTRL_REG, RADIO_SLEEP);
+
+    while (REG_GETF(CRG_TOP, SYS_STAT_REG, RAD_IS_UP) == 0x0);
+
+    REG_SETF(CRG_TOP, CLK_RADIO_REG, FTDF_MAC_ENABLE, 1);
+    REG_SETF(CRG_TOP, CLK_RADIO_REG, FTDF_MAC_DIV, 0);
+
+    phy_power_init();
+    hw_rf_system_init();
+    hw_rf_set_recommended_settings();
+    hw_rf_iff_calibration();
+
+    hw_rf_modulation_gain_calibration(0);  //RF MODE 0
+    hw_rf_dc_offset_calibration();
+
+    // ad_ftdf_init_phy_api();
+    FTDF_pib.CCAMode = 2;
+    ad_ftdf_init_phy_api_V2();
+    ad_ftdf_init_lmac();
+}
+
+
+ThreadError otPlatRadioEnable(otInstance *aInstance)
+{
+    First_Instace = aInstance;
+    (void)aInstance;
+    ThreadError error = kThreadError_None;
+    uint16_t DefaultChannel = DEFAULT_CHANNEL;
+    uint8_t value = 1;
+
+    FTDF_setValue(FTDF_PIB_RX_ON_WHEN_IDLE, &value); //Wake();
+    FTDF_setValue(FTDF_PIB_CURRENT_CHANNEL, &DefaultChannel); //SetChannel(channel);
+
+    uint32_t max_retries = 0;
+    FTDF_setValue(FTDF_PIB_MAX_FRAME_RETRIES, &max_retries);
+
+    FTDF_Bitmap32 options =  FTDF_TRANSPARENT_ENABLE_FCS_GENERATION;
+    options |= FTDF_TRANSPARENT_WAIT_FOR_ACK;
+    options |= FTDF_TRANSPARENT_AUTO_ACK;
+
+    FTDF_enableTransparentMode(FTDF_TRUE, options);
+    otPlatRadioSetPromiscuous(aInstance, false);
+
+    s_receive_frame.mPsdu = s_receive_message.mPsdu;
+    s_transmit_frame.mPsdu = s_transmit_message.mPsdu;
+    s_ack_frame.mPsdu = s_ack_message.mPsdu;
+    s_state = kStateReceive;
+    return error;
+}
+
+ThreadError otPlatRadioDisable(otInstance *aInstance)
+{
+    (void)aInstance;
+    s_state = kStateDisabled;
+    return kThreadError_None;
+}
+
+bool otPlatRadioIsEnabled(otInstance *aInstance)
+{
+    (void)aInstance;
+    return (s_state != kStateDisabled) ? true : false;
+}
+
+ThreadError otPlatRadioSleep(otInstance *aInstance)
+{
+    ThreadError error = kThreadError_Busy;
+    (void)aInstance;
+
+
+    if (s_state == kStateReceive && sleep_init_delay == 0)
+    {
+        sleep_init_delay = otPlatAlarmGetNow();
+        return kThreadError_None; //error;
+    }
+    else if ((otPlatAlarmGetNow() - sleep_init_delay) < 3000)
+    {
+        return kThreadError_None;    //error;
+    }
+
+    if (s_state == kStateSleep || s_state == kStateReceive)
+    {
+        error = kThreadError_None;
+        SED = true;
+        sleep_const_delay = otPlatAlarmGetNow();
+    }
+
+    return error;
+}
+
+ThreadError otPlatRadioReceive(otInstance *aInstance, uint8_t aChannel)
+{
+    ThreadError error = kThreadError_None;
+    (void)aInstance;
+
+#if DEBUG_LOG_ENABLE
+    hw_uart_write_buffer(HW_UART1, "\nR", 2);
+#endif
+    VerifyOrExit(s_state != kStateDisabled, error = kThreadError_Busy);
+    s_state = kStateReceive;
+    s_receive_frame.mChannel = aChannel;
+
+    uint32_t phyAckAttr;
+    phyAckAttr = 0x08 | ((aChannel - FTDF_OFFSET_CHANNEL) & 0xf) << 4 | (0 & 0x3) << 8;
```

Comment:
Why is there a `(0 & 0x3)`?  Isn't that just 0?  And bit-shifting a 0, that's also just 0